mainnet-contracts/src/PufferVaultV3.sol

```diff
@@ -0,0 +1,217 @@
+import { IDelegationManager } from "./interface/EigenLayer/IDelegationManager.sol";
+import { IWETH } from "./interface/Other/IWETH.sol";
+import { IPufferVaultV3 } from "./interface/IPufferVaultV3.sol";
+import { IPufferOracle } from "./interface/IPufferOracle.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { EnumerableMap } from "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { IConnext } from "./interface/Connext/IConnext.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IXERC20Lockbox } from "./interface/IXERC20Lockbox.sol";
+
+/**
+ * @title PufferVaultV3
+ * @dev Implementation of the PufferVault version 3 contract.
+ * @notice This contract extends the functionality of PufferVaultV2 with additional features for reward minting and bridging.
+ * @custom:security-contact security@puffer.fi
+ */
+contract PufferVaultV3 is PufferVaultV2, IPufferVaultV3 {
+    // The Connext contract on the origin domain.
+    IConnext public immutable _CONNEXT;
+    // The token to be paid on this domain.
+    IERC20 public immutable XTOKEN;
+    // The lockbox contract for xToken.
+    IXERC20Lockbox public immutable LOCKBOX;
+    // The destination domain ID for bridging.
+    uint32 public immutable _DESTINATION_DOMAIN;
+    // The address of the L2 reward manager.
+    address public immutable L2_REWARD_MANAGER;
+
+    // Slippage constant used in bridging transactions.
+    uint256 internal constant _SLIPPAGE = 0;
+
+    /**
+     * @notice Initializes the PufferVaultV3 contract.
+     * @param stETH Address of the stETH token contract.
+     * @param weth Address of the WETH token contract.
+     * @param lidoWithdrawalQueue Address of the Lido withdrawal queue contract.
+     * @param stETHStrategy Address of the stETH strategy contract.
+     * @param eigenStrategyManager Address of the EigenLayer strategy manager contract.
+     * @param oracle Address of the PufferOracle contract.
+     * @param delegationManager Address of the delegation manager contract.
+     * @param bridgingConstructorParams Constructor parameters for bridging.
+     */
+    constructor(
+        IStETH stETH,
+        IWETH weth,
+        ILidoWithdrawalQueue lidoWithdrawalQueue,
+        IStrategy stETHStrategy,
+        IEigenLayer eigenStrategyManager,
+        IPufferOracle oracle,
+        IDelegationManager delegationManager,
+        BridgingConstructorParams memory bridgingConstructorParams
+    ) PufferVaultV2(stETH, weth, lidoWithdrawalQueue, stETHStrategy, eigenStrategyManager, oracle, delegationManager) {
+        _CONNEXT = IConnext(bridgingConstructorParams.connext);
```

Comment:
Connext (now Everclear, they rebranded) is not the only bridge solution that supports this standard, maybe we should have an array of bridges, [everclear,jasonBridge,leaBridge]. And instead of relying on one bridge, our permissioned `mintAndBurn` could have a `bridgeIndex` parameter. That way, our product is somewhat decoupled from the dependency.